### PR TITLE
Read and Query Layers From Catalog

### PR DESCRIPTION
**Pull request by jbouffard**
This Pr updates the `read` and `query` methods so that they return either `SpatialTiledRasterRDD` or `TemporalTiledRasterRDD` in scala, and a `TiledRasterRDD` in python.

### CODE REVIEW

**Comment by jbouffard** [comment]
@echeipesh This is ready for review.

File: geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TiledRasterRDD.scala [hunk 1]
```diff
@@ -66,7 +66,7 @@ class SpatialTiledRasterRDD(
 }
 
 
-class SpatialTemporalTiledRasterRDD(
+class TemporalTiledRasterRDD(
```
Comment (echeipesh) [inline geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/TiledRasterRDD.scala]: Yeah, I buy this, we're already diverging from GT API a little and having a shorter names makes this a little easier to think about.

File: geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/io/LayerReaderFactory.scala [hunk 2]
```diff
@@ -66,26 +68,26 @@ abstract class FilteringLayerReaderWrapper()
...
-  ): (JavaRDD[Array[Byte]], String) = {
+  ): TiledRasterRDD[_] = {
     val id = LayerId(layerName, zoom)
     val valueClass = getValueClass(id)
 
     (keyType, valueClass) match {
       case ("SpatialKey", "geotrellis.raster.Tile") => {
         val result = layerReader.read[SpatialKey, Tile, TileLayerMetadata[SpatialKey]](id)
-        PythonTranslator.toPython(tileToMultiband[SpatialKey](result))
+        new SpatialTiledRasterRDD(MultibandTileLayerRDD(tileToMultiband[SpatialKey](result), result.metadata))
```
Comment (echeipesh) [inline geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/io/LayerReaderFactory.scala]: Did you decide against holding the zoom level field in `TiledLayerRDD` ?

Comment (jbouffard) [inline geopyspark-backend/geotrellis/src/main/scala/geopyspark/geotrellis/io/LayerReaderFactory.scala]: I was actually going to put that in a separate PR next. However, if you wanted it in this PR I can add it in.

**Pull request merged**